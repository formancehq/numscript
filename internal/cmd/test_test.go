package cmd

import (
	"bytes"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestShowDiff(t *testing.T) {
	var buf bytes.Buffer
	showDiff(
		&buf,
		map[string]any{
			"common": "ok",
			"x":      42,
		},
		map[string]any{
			"common": "ok",
			"x":      100,
		},
	)
	snaps.MatchSnapshot(t, buf.String())
}

func TestSingleTest(t *testing.T) {
	var out bytes.Buffer

	script := `
		send [USD/2 100] (
			source = @world
			destination = @dest
		)
	`

	specs := `
		{
			"testCases": [
				{
					"it": "tfailing",
					"expect.postings": [{
						"source": "wrong-source",
						"destination": "dest",
						"asset": "USD/2",
						"amount": 100
					}]
				},
				{
					"it": "tpassing",
					"expect.postings": [{
						"source": "world",
						"destination": "dest",
						"asset": "USD/2",
						"amount": 100
					}]
				}
			] 
		}
	`

	success := runRawSpecs(&out, &out, []rawSpec{
		{
			NumscriptPath:    "exmaple.num",
			SpecsPath:        "example.num.specs.json",
			NumscriptContent: script,
			SpecsFileContent: []byte(specs),
		},
	})

	require.False(t, success)

	snaps.MatchSnapshot(t, out.String())
}
