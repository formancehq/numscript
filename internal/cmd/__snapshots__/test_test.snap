
[TestShowDiff - 1]
[90m  {[0m
[90m    "common": "ok",[0m
[32m-   "x": 42[0m
[31m+   "x": 100[0m
[90m  }[0m

---

[TestSingleTest - 1]
[31m❯[0m exmaple.num [90m(2 tests | [31m1 failed[0m[90m)[0m
  [31m×[0m tfailing


[1m[97m[41m FAIL [0m[97m[0m[1m[0m[31m example.num.specs.json > tfailing[0m

[32m- Expected[0m
[31m+ Received
[0m
[4mexpect.postings[0m

[90m  [[0m
[90m    {[0m
[32m-     "source": "wrong-source",[0m
[31m+     "source": "world",[0m
[90m      "destination": "dest",[0m
[90m      "amount": 100,[0m
[90m      "asset": "USD/2"[0m
[90m    }[0m
[90m  ][0m

[90m Test files [0m [1m[91m1 failed[0m[1m[0m [90m(1)[0m
[90m      Tests [0m [1m[91m1 failed[0m[1m[0m[90m | [0m[1m[92m1 passed[0m[1m[0m [90m(2)[0m

---
